apps/backend/src/reconciliation/pos-reconciliation.service.ts

```diff
@@ -84,85 +67,71 @@ export class PosReconciliationService {
       PosReconciliationService.name
     );
 
-    const roundOneMatches: {
-      payment: PaymentEntity;
-      deposit: POSDepositEntity;
-    }[] = this.matchPosPaymentToPosDeposits(
-      pendingPayments.filter(
-        (itm) =>
-          itm.status === MatchStatus.PENDING ||
-          itm.status === MatchStatus.IN_PROGRESS
-      ),
-      pendingDeposits.filter(
-        (itm) =>
-          itm.status === MatchStatus.PENDING ||
-          itm.status === MatchStatus.IN_PROGRESS
-      ),
-      PosHeuristicRound.ONE
-    );
-
-    this.appLogger.log(
-      `MATCHES - ROUND ONE ${roundOneMatches.length}`,
-      PosReconciliationService.name
+    // Put deposits into a dictionary, first layer keys are amount, second layer keys are dates
+    // This makes it more efficient to find than looping through all deposits
+    const locationPosDepositsDictionary = pendingDeposits.reduce(
```

Comment:
Should we be grouping these first by date, and then by amount? We want to make sure we're getting 1:1 matching for the first two heuristic rounds. (I might be missing something still reviewing)